hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/MapWarmer.java

```diff
@@ -0,0 +1,97 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.mono.sigs.order;
+
+import com.hedera.node.app.service.mono.context.properties.GlobalDynamicProperties;
+import com.hedera.node.app.service.mono.state.migration.AccountStorageAdapter;
+import com.hedera.node.app.service.mono.state.migration.TokenRelStorageAdapter;
+import com.hedera.node.app.service.mono.state.migration.UniqueTokenMapAdapter;
+import com.hedera.node.app.service.mono.state.virtual.EntityNumVirtualKey;
+import com.hedera.node.app.service.mono.state.virtual.UniqueTokenKey;
+import com.hedera.node.app.service.mono.state.virtual.UniqueTokenValue;
+import com.hedera.node.app.service.mono.state.virtual.entities.OnDiskAccount;
+import com.hedera.node.app.service.mono.state.virtual.entities.OnDiskTokenRel;
+import com.swirlds.virtualmap.VirtualKey;
+import com.swirlds.virtualmap.VirtualMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.function.Supplier;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class MapWarmer {
+
+    private static final Logger log = LoggerFactory.getLogger(MapWarmer.class);
+
+    private VirtualMap<EntityNumVirtualKey, OnDiskAccount> accounts = null;
+    private VirtualMap<UniqueTokenKey, UniqueTokenValue> nfts = null;
+    private VirtualMap<EntityNumVirtualKey, OnDiskTokenRel> tokenRels = null;
+
+    private final Supplier<AccountStorageAdapter> accountsStorageAdapter;
+    private final Supplier<UniqueTokenMapAdapter> nftsAdapter;
+    private final Supplier<TokenRelStorageAdapter> tokenRelsAdapter;
+
+    private final Executor threadpool;
+
+    @Inject
+    public MapWarmer(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            GlobalDynamicProperties globalDynamicProperties) {
+        this.accountsStorageAdapter = accountsStorageAdapter;
+        this.nftsAdapter = nftsAdapter;
+        this.tokenRelsAdapter = tokenRelsAdapter;
+        this.threadpool =
+                Executors.newFixedThreadPool(globalDynamicProperties.cryptoTransferWarmThreads());
+    }
+
+    public void warmAccount(EntityNumVirtualKey accountId) {
+        if (accounts == null) {
+            var accountsStorage = accountsStorageAdapter.get();
+            accounts = accountsStorage.areOnDisk() ? accountsStorage.getOnDiskAccounts() : null;
+        }
```

Comment:
I don't like checking for `null` here every time `warm` is called, but otherwise there's a circular dependency between the dagger `StateModule` and this module (`SigModule`) and the adapters weren't loaded at construction time.